From a282a6b39a5606dc42e7481d8b56675c937c0136 Mon Sep 17 00:00:00 2001
From: Felix Hoehle <fhoehle@users.noreply.github.com>
Date: Mon, 9 Feb 2026 14:24:36 +0100
Subject: [PATCH] replace beta releases of numba and llmvlite

---
 pyproject.toml | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index d9c9fdd..493b31b 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -22,8 +22,7 @@ dependencies = [
   'tqdm>=4.27.0',
   'packaging>20.9',
   'slicer==0.0.8',
-  'numba==0.63.0b1; python_version >= "3.14"',  # todo: remove once numba officially releases support for 3.14
-  'llvmlite==0.46.0b1; python_version >= "3.14"',  # todo: remove once llvmlite officially releases support for 3.14
+  'numba>=0.63.0; python_version >= "3.14"',
   'numba>=0.54; python_version < "3.14"',
   'cloudpickle',
   'typing-extensions',
-- 
2.52.0

